pkg/stanza/operator/config.go

```diff
@@ -25,7 +26,7 @@ func NewConfig(b Builder) Config {
 type Builder interface {
 	ID() string
 	Type() string
-	Build(*zap.SugaredLogger) (Operator, error)
```

Comment:
It's unfortunate, but because this is exported, we'll have to deprecate it as is an replace it with another version of the function. This is a lot to ask so I'll take a pass at it and open a PR for it.

Reply:
Ah ok I see. Thank's for handling this. Let me know if there is anything I could help with.

Reply:
I've made some progress on this but the migrating to a new interface leads to a lot of changes throughout the codebase. I have a few more things to work through next week and then can start peeling off PRs & hopefully isolate some things which can be done in parallel.

Reply:
Thank's for the update. I will have a look next week and share back any findings.

Reply:
I didn't manage to achieve a lot beyond verifying the issue and trying to massage https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31664 to work.
Even if I managed to resolve the compiling errors, then on runtime the Collector fails to unmarshal configs
based on the new approach :disappointed::

```console=
Error: failed to get config: cannot unmarshal the configuration: 1 error(s) decoding:

* error decoding 'receivers': error reading configuration for "filelog": 3 error(s) decoding:

* 'operators[0]' expected type 'operator.Identifiable', got 'map[string]interface {}'
* 'operators[1]' expected type 'operator.Identifiable', got 'map[string]interface {}'
* 'operators[2]' expected type 'operator.Identifiable', got 'map[string]interface {}'
2024/04/11 15:58:24 collector server run finished with error: failed to get config: cannot unmarshal the configuration: 1 error(s) decoding:
```

Based on the fact that the main issue seems to be the `fundamentally different pattern to unmarshaling operator configs vs component configs because of how identity is managed (inside vs outside)` I wonder if we should take a step back and think of fixing this at first place?
Not sure if this is technically doable though.